lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/service/LessonEntity.java

```diff
@@ -119,6 +119,7 @@ public interface LessonEntity {
     public boolean isUsable();
     // only assignments
     public int getTypeOfGrade();
+    public int getSubmissionType();
```

Comment:
I'd add a default to this. It looks like you're returning 0 from the non implemented tools, so just do that here.
default public int getSubmissionType() {
    return 0;
}

You can then remove them from the non assignment lessons entities.

Reply:
Didn't even occur to me to use default to cut down on changes needed. Thanks!